test/sqlite3_monkey_patch.rb

```diff
@@ -0,0 +1,29 @@
+=begin
+This monkey patch is necessary because the SQLite3Adapter is smart enough to encode strings
+that come in with Encoding::ASCII-8BIT to Encoding::UTF-8 but others aren't.
+So to mimic the worst case, for example using a bad oracle adapter, we monkey patch the
+SQLite3Adapter to just return the string instead of trying to encode it using utf8.
+=end
```

Comment:
Great comment! I think any 5.x (and up) adapter would exhibit this behavior to the type changes. If so, might be worth calling that out? https://technology.customink.com/blog/2015/01/06/activerecord-42s-type-casting/